pygdf/dataframe.py

```diff
@@ -318,6 +319,36 @@ def copy(self):
         df._cols = self._cols.copy()
         return df
 
+    def _sanitize_columns(self, col):
+        """Sanitize pre-appended
+           col values
+        """
+        if len(self) == 0 and len(self.columns) > 0:
+            series = Series(col)
+            data = np.zeros(len(series.index))
```

Comment:
This will execute on the CPU and for larger dataframes end up taking a long time to generate the array on the CPU and then transfer to the GPU, correct? Is there a way we could generate the data on the GPU instead?

Reply:
Yes, correct. the array is being initialized in CPU. There is a way to initialize it on GPU. I will take a look at it.

Reply:
I made the changes. Seems good now.